### PR TITLE
CI MacOS 12 Update

### PR DESCRIPTION
**Pull request by jeremylt**

### CODE REVIEW

**Comment by jeremylt** [comment]
I want to let CI finish, but this should fix the MacOS error

**Review by jedbrown — APPROVED** [review]

**Comment by jeremylt** [comment]
Ready to merge - since this changes the 'required' test runs, it requires admin privileges to squash+merge @jedbrown

File: .github/workflows/c-fortran-test-hardware.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ jobs:
   test:
     strategy:
       matrix:
-        os: [ubuntu-20.04]
-        compiler: [gcc-9]
+        os: [ubuntu-22.04]
+        compiler: [gcc-11]
         arch: [aarch64, ppc64le]
         distro: [ubuntu20.04]
```
Comment (jedbrown) [inline .github/workflows/c-fortran-test-hardware.yml:13]: Seems weird to use distro 20.04 with os 22.04?

**Review by jedbrown — COMMENTED** [review]
Should we update the python jobs too?

File: .github/workflows/c-fortran-test-hardware.yml (hunk 1 above)
Comment (jeremylt) [reply .github/workflows/c-fortran-test-hardware.yml:13]: As I understand it, the parent container will use 22.04 and run the tests in a child image with 20.04

**Comment by jeremylt** [comment]
> Should we update the python jobs too?


Might as well

**Comment by jedbrown** [comment]
Yeah, I was just looking through those required checks and it's a somewhat odd mix. Maybe that's fine/good, but I don't know what it means to use a different `distro` versus `os`.

File: .github/workflows/c-fortran-test-hardware.yml (hunk 1 above)
Comment (jeremylt) [reply .github/workflows/c-fortran-test-hardware.yml]: Bumping the run-on-hardware action version let me sync those two

**Comment by jedbrown** [comment]
Looks good to squash-merge now.

**Comment by jeremylt** [comment]
Agreed, good to squash+merge

**Pull request merged**